common/src/chain/config/emission_schedule.rs

```diff
@@ -0,0 +1,336 @@
+use crate::primitives::{Amount, BlockHeight};
+use std::time::Duration;
+
+// TODO Move to a separate module
+/// Represents a certain amount of MLT.
+#[derive(Clone, Copy, Eq, PartialEq, PartialOrd, Ord, Debug)]
+pub struct Mlt(Amount);
+
+impl Mlt {
+    /// Number of decimal digits used to represent MLT
+    pub const DECIMALS: u8 = 11;
+    /// Number of atoms in 1 MLT
+    pub const ATOMS_PER_MLT: u128 = 10u128.pow(Self::DECIMALS as u32);
+    /// Zero MLTs
+    pub const ZERO: Self = Self(Amount::from_atoms(0));
+    /// Maximum representable amount of MLTs
+    pub const MAX: Self = Self(Amount::MAX);
+
+    /// Construct from the number atomic units
+    pub const fn from_atoms(n: u128) -> Self {
+        Self(Amount::from_atoms(n))
+    }
+
+    /// Construct from the number of MLTs
+    pub const fn from_mlt(n: u64) -> Self {
```

Comment:
While I understand why you're using u64 here, but it should basically be the int type of Amount (it's public, btw). Then the result should be an Option or Result.

Related to that, I understand that you're using const fn, but this is wrong as it breaks the requirements. Remember, when you brought this up the first time, I mentioned the issue with moving from strings. Even though the Mlt type is a good idea, there has to be a way to read this from strings (or string -> u64 -> Amount, whichever you prefer), because we will want to establish a schedule in the functional tests as a command line argument. If you're writing this as a generic MLT struct, then there has to be a way to initialize MLT at runtime with MLT amount. This function seems appropriate for that. It's a nice touch to have this work with static assert, but this has to always work at runtime as well, despite it not having a specific use right now.

To summarize: Either add another function for runtime, or make this one do that. I think this being a const fn is unnecessary. Just do the Amount multiplication and return an Option and expect/unwrap on usage.

Reply:
Regarding `const`-ness here, `const fn` works at runtime too. Annotating the function `const` just places some extra restrictions on it so it can be used in `const` contexts (like initializing some constants below) but executing it at runtime is still possible.

Reply:
The static assert only refers to constants and types, not to the input value. It could well be outside of this function but I decided to put it inside because it is related to the argument made in the comment. It does not hinder the ability to run the function at runtime in any way.